instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py

```diff
@@ -284,12 +287,17 @@ def wrapped_connection(
     ):
         """Add object proxy to connection object."""
         connection = connect_method(*args, **kwargs)
-        self.get_connection_attributes(connection)
+        self.get_connection_attributes(connection=connection, kwargs=kwargs)
         return get_traced_connection_proxy(connection, self)
 
-    def get_connection_attributes(self, connection):
-        # Populate span fields using connection
+    def get_connection_attributes(self, connection, kwargs=None):
+        # Populate span fields using kwargs and connection
         for key, value in self.connection_attributes.items():
+            # First set from kwargs
+            if kwargs and value in kwargs:
+                self.connection_props[key] = kwargs.get(value)
```

Comment:
Thanks for putting in this PR!

> 3. Uses more sources for the db.user, net.peer.name, and net.peer.port span attributes.

Could some tests be added to show expectations for this?